### PR TITLE
Use saturating ops when passing computed clip value to gecko (fix bug 1384150)

### PR DESCRIPTION
**Pull request by Manishearth**
https://bugzilla.mozilla.org/show_bug.cgi?id=1384150

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17856)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -3723,14 +3723,14 @@ fn static_assert() {
                 }
 
                 if let Some(bottom) = rect.bottom {
-                    self.gecko.mClip.height = bottom.0 - self.gecko.mClip.y;
+                    self.gecko.mClip.height = bottom.0.saturating_sub(self.gecko.mClip.y);
```
Comment (emilio) [inline components/style/properties/gecko.mako.rs]: Why not `(bottom - Au(self.gecko.mClip.y)).0`? Seems more consistent if we change the the behavior of `Au`.

Comment (emilio) [inline components/style/properties/gecko.mako.rs]: (Same below)

**Comment by Manishearth** [comment]
done

**Pull request merged**